### ISSUE TITLE
can't tag ELBs

### ISSUE THREAD

**Issue by smiller171** [ansible/ansible#18927]
_From @smiller171 on December 3, 2014 18:34_

##### Issue Type:

Feature Idea
##### Component Name:

ec2_elb_lb module
##### Ansible Version:

1.8.1
##### Environment:

Managing EC2 from generic.
##### Summary:

Currently there is no way to tag an ELB because it does not have a resource_id, so it can't be tagged by ec2_tag.
##### Steps To Reproduce:

I think that you should either have a "tags" parameter in the ec2_elb_lb module (like instance_tags in the ec2 module) or allow for the use of the resource name in ec2_tag.
##### Expected Results:

I expect to be able to add tags to ELBs by using either a parameter in the ec2_elb_lb module or using the ec2_tag module.
##### Actual Results:

Currently you can't tag ELBs from Ansible.

_Copied from original issue: ansible/ansible-modules-core#439_

**Comment by smiller171** [comment]
_From @bobby on January 19, 2015 18:45_

+1, it would be nice to be able to uniformly tag all AWS resources provisioned via Ansible.

**Comment by smiller171** [comment]
_From @nicolas-g on February 2, 2015 14:14_

+1 , tagging has become really important for cost reporting, administration and automation ...

**Comment by smiller171** [comment]
_From @sebest on February 24, 2015 1:22_

+1 here

**Comment by smiller171** [comment]
_From @evitorbarros on March 9, 2015 20:0_

+1

**Comment by smiller171** [comment]
_From @ryanwalls on March 27, 2015 22:23_

+1

**Comment by smiller171** [comment]
_From @k3ypad on March 29, 2015 9:12_

Email sent to ansible devel list - should have something soon:
https://groups.google.com/forum/#!topic/ansible-devel/FJZ8eOqhfOc

**Comment by smiller171** [comment]
_From @eproxus on June 11, 2015 10:16_

What's the current status? Still missing boto support?

**Comment by smiller171** [comment]
@eproxus yup, still missing. I'm tagging ELBs in my playbooks using the shell module to run AWSCLI commands

**Comment by smiller171** [comment]
_From @eproxus on June 11, 2015 12:21_

Ah, do you have a snippet you could share? :smile:

**Comment by smiller171** [comment]
```
- name: tag ELB
  shell: "aws --region us-east-1 elb add-tags --load-balancer-names {{elbname}} --tags Key=stack,Value={{stack}} Key=layer,Value={{layer}} Key=owner,Value={{owner}}"
```

**Comment by smiller171** [comment]
_From @eproxus on June 11, 2015 14:20_

@smiller171 Awesome, thanks!

**Comment by smiller171** [comment]
no problem @eproxus. I think Ansible 2.0 is supposed to switch to Boto3, which does have tagging support for ELBs

**Comment by smiller171** [comment]
_From @s0enke on July 13, 2015 10:16_

> no problem @eproxus. I think Ansible 2.0 is supposed to switch to Boto3, which does have tagging support for ELBs

@smiller171 Do you have any particular information about that? I could not find any hints that Ansible2 is going to be ported to Boto3.

**Comment by smiller171** [comment]
After much searching, I found nothing. I have suspicions about where this information came from, but don't recall for sure.

**Comment by smiller171** [comment]
_From @groorj on January 4, 2016 22:48_

+1

**Comment by smiller171** [comment]
_From @welex91 on January 5, 2016 18:34_

I've created a PR in case someone would like to test this: https://github.com/ansible/ansible-modules-extras/pull/1454

**Comment by smiller171** [comment]
_From @Constantin07 on January 8, 2016 10:36_

+1

**Comment by smiller171** [comment]
_From @welex91 on January 19, 2016 18:18_

Another PR: https://github.com/ansible/ansible-modules-core/pull/2844 this time patching ec2_elb_lb, based on helpful feedback from @jonhadfield and @Constantin07

**Comment by smiller171** [comment]
_From @nand0p on February 25, 2016 17:0_

https://github.com/ansible/ansible-modules-extras/pull/1738 boto3 mod.  returns changed only if elb tags are changed.

**Comment by smiller171** [comment]
_From @nand0p on March 1, 2016 21:29_

https://github.com/ansible/ansible-modules-core/pull/3163 updates ec2_elb_lb module to support add + modify tags

**Comment by smiller171** [comment]
_From @shay-li on March 22, 2016 15:26_

+1

**Comment by smiller171** [comment]
_From @ericcitaire on April 29, 2016 15:36_

https://github.com/ansible/ansible-modules-extras/pull/2133 adds a new module ec2_elb_tag.

**Comment by smiller171** [comment]
_From @ansibot on August 1, 2016 3:49_

@jsdalton, @rickmendes, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by smiller171** [comment]
_From @ansibot on September 13, 2016 2:0_

@jsdalton, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by smiller171** [comment]
_From @ansibot on October 6, 2016 16:44_

@jsdalton, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by smiller171** [comment]
_From @ansibot on December 7, 2016 18:8_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
cc @willthames
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @jillr @s-hertel @tremble
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/cloud/amazon/ec2_elb_lb.py -> https://galaxy.ansible.com/amazon/aws

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
